Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -533,6 +542,26 @@ extension SQLiteBookmarks {
             (deleteStructure, args),
         ])
     }
+    
+    public func updateBookmarkByGUID(new: BookmarkItem) -> Success {
+        let args: Args = [
+            NSDate.nowNumber(),
+            new.url,
+            new.title,
+            SyncStatus.Changed.rawValue,
+            new.guid
+            ]
+        
+        let update =
+            "UPDATE \(TableBookmarksLocal) SET" +
+                " local_modified = ?" +
+                ", bmkUri = ?" +
+                ", title = ?" +
+                ", sync_status = ?" +
+                " WHERE guid = ?"
+        
+        return self.db.run([(update,args)])
```

Comment:
You're correct, Steph.

Take a look at the existing code for removing a bookmark. You'll see it deletes the local bookmark if it exists, marking the parent as changed. If it's not a local bookmark, it overrides the mirror contents, again updating metadata. The same thing needs to happen here, but even more so: the bookmark you're editing might be a read-only shadow, a merged item in the mirror, a merged item that's already overridden by a changed local version, or a brand-new local item that hasn't been synced yet.

You should add tests for each of these situations, just like Steph's `isEditable` tests.